common/txmgr/confirmer.go

```diff
@@ -1259,7 +1259,12 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) sen
 
 // ResumePendingTaskRuns issues callbacks to task runs that are pending waiting for receipts
 func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) ResumePendingTaskRuns(ctx context.Context, head types.Head[BLOCK_HASH]) error {
-	receiptsPlus, err := ec.txStore.FindTxesPendingCallback(ctx, head.BlockNumber(), ec.chainID)
+	latest := head.BlockNumber()
+	var finalized int64
+	if finalizedHead := head.LatestFinalizedHead(); finalizedHead != nil {
+		finalized = finalizedHead.BlockNumber()
+	}
```

Comment:
I don't think we should retrieve the finalized head from the latest head here. IIRC this doesn't provide us repeatable read guarantees. Think we can just pass in the finalized head we retrieved directly from the headtracker [here](https://github.com/smartcontractkit/chainlink/blob/816b25c00122003e14e3d214d9de7a6e58b3323e/common/txmgr/confirmer.go#L308) into the `ResumePendingTaskRuns` method.

Reply:
Sure, but then why are we passing around a complex Head interface with methods that are not valid for use?

Reply:
I remember there was a thread about this but can't remember what was decided. @dhaidashenko Did we ever find a resolution for this like removing this method? Or if there was a reason we kept it?

Reply:
Well this method can just accept int64 to avoid the issue.